protocol/block.rei

```diff
@@ -21,6 +21,7 @@ let verify: (~signature: Signature.t, t) => bool;
 let genesis: t;
 let produce:
   (
+    ~state_root_hash: BLAKE2B.t,
```

Comment:
This signature is not right, the function already receives the state, it should be able access the state root hash available.

Reply:
And especially this allows to produce a block with the wrong state_root_hash

Reply:
Fixed here: https://github.com/marigold-dev/deku/pull/195/commits/08de202b8d9b9ea950c8cd2a29ef3a108bcbc4e5